### ISSUE TITLE
skydns is killed due to liveness probe failure or crashloop 

### ISSUE THREAD

**Issue by dchen1107** [GoogleCloudPlatform/kubernetes#10513]
This might be the root cause of another issue I saw related to heapster: #10511. 

I had cluster/kube-down.sh and hack/dev-build-and-up.sh for several times (roughtly 5 times) this afternoon, and ran into this twice:

1) I noticed that kube-dns with some abnormal restart count:

```
$ cluster/kubectl.sh get pods
NAME                                           READY     REASON    RESTARTS   AGE
fluentd-cloud-logging-kubernetes-minion-d615   1/1       Running   0          6m
fluentd-cloud-logging-kubernetes-minion-ds5p   1/1       Running   0          7m
fluentd-cloud-logging-kubernetes-minion-ed39   1/1       Running   0          6m
fluentd-cloud-logging-kubernetes-minion-srpn   1/1       Running   0          7m
kube-dns-v4-af0jm                              3/3       Running   3          8m
monitoring-heapster-v5-pc3el                   0/1       Running   9          8m
monitoring-influx-grafana-v1-ejij9             2/2       Running   0          8m
```

2) kubectl describe pods shows skydns is killed due to liveness probe:

```
  Mon, 29 Jun 2015 16:37:20 -0700	Mon, 29 Jun 2015 16:44:50 -0700	16	{kubelet kubernetes-minion-ed39}	spec.containers{skydns}			unhealthy		Liveness probe failed: nslookup: can't resolve 'kubernetes.default.svc.cluster.local'
```

3) checked docker container --skydns log:

```
$ cluster/kubectl.sh logs  kube-dns-v4-af0jm skydns --previous
2015/06/29 23:45:50 skydns: falling back to default configuration, could not read from etcd: 100: Key not found (/skydns) [2]
2015/06/29 23:45:50 skydns: ready for queries on cluster.local. for tcp://0.0.0.0:53 [rcache 0]
2015/06/29 23:45:50 skydns: ready for queries on cluster.local. for udp://0.0.0.0:53 [rcache 0]
```

cc/ @thockin 

I believe this is a new regression we have in our HEAD.

**Comment by a-robinson** [comment]
Look at the kube2sky logs -- I'm pretty confident it'll have the same problem that Heapster is having in #10511, namely that it can't find the secret it expects (/var/run/secrets/kubernetes.io/serviceaccount/token)

**Comment by thockin** [comment]
This worked on Friday - I brought up a cluster and it worked?

**Comment by a-robinson** [comment]
It's been flaky the last couple days. The dns (and heapster) failures seem to be just be symptoms of a serviceaccount problem

**Comment by a-robinson** [comment]
Although the serviceaccount e2e hasn't had any flakes at all...

**Comment by dchen1107** [comment]
Closed it as dup of #10507
